### PR TITLE
evm: Reduce contract size of Manager

### PR DESCRIPTION
**Pull request by djb15**
This set of commits gives us just over 2kB of breathing room without touching the invariants. Have a look at the commit messages to see how much each set of changes saves us.

### CODE REVIEW

**Review by RahulMaganti47 — DISMISSED** [review]

File: evm/src/libraries/EndpointHelpers.sol [hunk 1]
```diff
@@ -35,3 +35,13 @@ function isFork(uint256 evmChainId) view returns (bool) {
...
+
+// @dev Count the number of set bits in a uint64
+function countSetBits(uint64 x) pure returns (uint8 count) {
+    while (x != 0) {
+        x &= x - 1;
+        count++;
+    }
+
+    return count;
+}
```
Comment (RahulMaganti47) [inline evm/src/libraries/EndpointHelpers.sol:47]: can we move this to a separate utils library or rename EndpointHelpers -> NTTUtils.sol

File: evm/src/Manager.sol [hunk 2]
```diff
@@ -779,24 +769,13 @@ contract Manager is
...
-
     /// ============== INVARIANTS =============================================
 
     /// @dev When we add new immutables, this function should be updated
     function _checkImmutables() internal view override {
         assert(this.token() == token);
         assert(this.mode() == mode);
         assert(this.chainId() == chainId);
-        assert(this.evmChainId() == evmChainId);
         assert(this.rateLimitDuration() == rateLimitDuration);
```
[the remark this answers is not in the document]
Comment (djb15) [reply evm/src/Manager.sol:779]: This is called during migrations from `Implementation.sol`

File: evm/src/libraries/EndpointHelpers.sol (hunk 1 above)
Comment (djb15) [reply evm/src/libraries/EndpointHelpers.sol:47]: I've opened https://github.com/wormhole-foundation/example-native-token-transfers/issues/183 to do this, distracts from the goal of this PR imo

File: evm/src/Manager.sol (hunk 2 above)
[the remark this answers is not in the document]
Comment (RahulMaganti47) [reply evm/src/Manager.sol:779]: ah yea. do you think we need this check? what are your thoughts on removing it?

[the remark this answers is not in the document]
Comment (kcsongor) [reply evm/src/Manager.sol:779]: 100% keep it

**Review by RahulMaganti47 — DISMISSED** [review]

**Review by RahulMaganti47 — APPROVED** [review]

**Pull request merged**